### PR TITLE
[Routing] Move configuration to PHP

### PR DESCRIPTION
**Pull request by phamuyentri**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | [#37186](https://github.com/symfony/symfony/issues/37186)
| License       | MIT

Moving routing configurations in FrameworkBundle to PHP

### CODE REVIEW

**Review by Tobion — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php [hunk 1]
```diff
@@ -30,11 +31,11 @@ class RedirectController
     private $httpPort;
     private $httpsPort;
 
-    public function __construct(UrlGeneratorInterface $router = null, int $httpPort = null, int $httpsPort = null)
+    public function __construct(RequestContext $requestContext, UrlGeneratorInterface $router = null)
```
Comment (phamuyentri) [inline src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php:34]: This change to cover the config in [router.xml:120](https://github.com/symfony/symfony/pull/37280/files#diff-a54ffdcb2ee5f6c0c06492fabab70827L120). Many thanks if you have a better way to do it.

Comment (Tobion) [reply src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php:34]: you can achieve the same in the php config. you need to use `inline_service`

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.php [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        ->set('routing.loader', DelegatingLoader::class)
+            ->public()
+            ->args([
+                service('routing.resolver'),
+                [], // Default options
+                [], // Default requirements
+            ])
+
+        ->set('router.default', Router::class)
+            ->tag('monolog.logger', ['channel' => 'router'])
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.php:92]: please move `tag` after `args` and `call`

**Review by Tobion — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (phamuyentri) [reply src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php]: Reverted.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.php [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+use Symfony\Component\Routing\Matcher\Dumper\CompiledUrlMatcherDumper;
+use Symfony\Component\Routing\Matcher\ExpressionLanguageProvider;
+use Symfony\Component\Routing\Matcher\UrlMatcherInterface;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\RequestContextAwareInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+return static function (ContainerConfigurator $container) {
+    $container->parameters()->set('router.request_context.host', 'localhost');
+    $container->parameters()->set('router.request_context.scheme', 'http');
```
[the remark this answers is not in the document]
Comment (phamuyentri) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.php]: Updated.

File: src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php (hunk 1 above)
Comment (phamuyentri) [reply src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php]: Thanks @Tobion. Updated.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.php (hunk 2 above)
Comment (phamuyentri) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.php]: Moved down all the tags.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.php [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+use Symfony\Component\Routing\Matcher\UrlMatcherInterface;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\RequestContextAwareInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+return static function (ContainerConfigurator $container) {
+    $container->parameters()
+        ->set('router.request_context.host', 'localhost')
+        ->set('router.request_context.scheme', 'http')
+        ->set('router.request_context.base_url', '');
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.php:43]: semicolon should be on the next line

**Review by Tobion — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.php (hunk 4 above)
Comment (phamuyentri) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.php]: Updated.

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @phamuyentri.

**Pull request merged**